### DISCUSSION TITLE
Dynamic values in body

### DISCUSSION THREAD

**Discussion by Arctomachine** [usebruno/bruno#1476] [Q&A] [answered]
I want to set dynamic values to json body. But I also want to keep the rest of body from manual input. How to achieve it?
Example:
```js
{
...
"created_at": Date.now(),
"id": crypto.randomUUID(),
"text": btoa(Math.random())
}
```
I tried bru.setVar, but it says variable is not defined. And when manually defining its value, it is not overwritten by script.
Also tried req.body.id = ... and req.setBody(...), but it has no effect over manually typed body.
And scripts give error about crypto, btoa

**Comment by rusak47** [comment]
I'm totally new to Bruno and can't figure out how to do this. It's a very useful feature, and Talend API Tester allows multiple expressions (albeit limited) to generate random data.
![talend_api_tester_expression_builder](https://github.com/usebruno/bruno/assets/8426197/972cc2f9-dece-4744-8d6b-22f3e249d9a7)

**Reply by rusak47** [reply]
Placing a valid JS script into a variable doesn't work - it's inserted as plain text into the body. In the tooltip for the variables field, I see a note that a JS Literal Template can be used, but I haven't figured out how to use it yet. for example to run ```(Math.random() + 1).toString(36).substring(7)```

https://www.youtube.com/watch?v=pu1WNjgdD3s

**Reply by rusak47** [reply]
Ok, it's actually quite simple:

1. put a pre-request variable with dummy value (e.g. randomStringVal = null or whatever)
2. place variable into body as {{randomStringVal}}
3. add a pre-request script that changes its value
e.g.
```let randomString = (Math.random() + 1).toString(36).substring(7);
bru.setVar("randomStringVal",randomString)```

https://www.youtube.com/watch?v=DYjSG4CPjo8

**Comment by aalmazan** [comment, marked as answer]
Over a year, but for those still wondering, there are dynamic variables similar to how Postman does it -- https://docs.usebruno.com/testing/script/dynamic-variables

So in your response JSON in Bruno you can do something like:
```json
{ 
    "name": "{{$randomFullName}}"
}
```

**Reply by Rich-Hopkins** [reply]
That's interesting - thanks for sharing it.

Now my suggestion would be, having seen the random lorem at the bottom, is to implement random bacon. 😂
http://www.baconipsum.com

Which makes me think that I need to implement more bacon in my test database entries for my coworkers to find. Picking on the QA guy can get tiresome. 😂
